File: 1-Dimension-DP/fib.js
function climbStairs(n) {
    if (n <= 2){ return 1 };
    let one = 1, two = 1;

    for (let i = 0; i < n - 1; i++) {
        const temp = one;
        one = one + two;
        two = temp;
    }
    return one;
}

console.log(climbStairs(5));

File: 1-Dimension-DP/fib.cs
using System;

class fib
{
    static int ClimbStairs(int n)
    {
        if (n <= 2)
        {
            return 1;
        }

        int one = 1, two = 1;

        for (int i = 0; i < n; i++)
        {
            int temp = one;
            one = one + two;
            two = temp;
        }

        return one;
    }

    static void Main(string[] args)
    {
        Console.WriteLine(ClimbStairs(5));
    }
}